### PR TITLE
Skip SmoothQuant creation if alpha is negative

### PR DESCRIPTION
**Pull request by nikita-savelyevv**
### Changes

Don't create Smooth Quant algorithm if alpha parameter is negative.

### Reason for changes

When running AA with `tune_hyperparams=True` and `smooth_quant_alpha<0`, it fails here https://github.com/openvinotoolkit/nncf/blob/develop/nncf/quantization/algorithms/post_training/algorithm.py#L168 because SQ had been created. This fix makes it so SQ is not created if it won't be used.

### CODE REVIEW

File: nncf/quantization/algorithms/smooth_quant/algorithm.py [hunk 1]
```diff
@@ -63,6 +63,10 @@ def __init__(self, subset_size: int = 300, inplace_statistics: bool = True, alph
         :param alpha: The parameter that regulates the calculation of the scale.
             The default value is 0.95. Negative value switches off the algorithm.
         """
+
+        if alpha < 0:
+            raise RuntimeError("Smooth Quant algorithm does not support negative alpha parameter!")
```
Comment (KodiaqQ) [inline nncf/quantization/algorithms/smooth_quant/algorithm.py:68]: I don't know, maybe should we keep `if self._alpha < 0:` conditions or do not add `RuntimeError`. @alexsu52, what do you think?

**Comment by nikita-savelyevv** [comment]
run pytorch pre-commit tests
run tensorflow pre-commit tests

File: nncf/quantization/algorithms/smooth_quant/algorithm.py (hunk 1 above)
Comment (alexsu52) [reply nncf/quantization/algorithms/smooth_quant/algorithm.py:68]: LGTM. @AlexKoff88, do you have a comment?

Comment (AlexKoff88) [reply nncf/quantization/algorithms/smooth_quant/algorithm.py:68]: I am ok with this too.

**Review by AlexKoff88 — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

**Review by KodiaqQ — APPROVED** [review]

**Pull request merged**